[CTG] replace load_yamls with load_yaml

trdthg opened the pull request:
## Description

seems not been updated synchronously when the function, variable were modified

UmerShahidengr commented:
@trdthg we would like to keep the naming convention of `load_yaml` to be consistent everywhere. Having two functions with names `load_yaml` and `load_yamls` will create massive confusion.
I can see there are two instances of `load_yamls` used which can be replaced with `load_yaml` instead of adding more callings of `load_yamls`.
My recommendation would be to removed `load_yamls` and replace it with `load_yaml` everywhere to keep everything consistent.

trdthg committed:
[CTG] replace load_yamls with load_yaml

- keep only one load_yaml functon to aviod confusion
- load_yaml will determine whether to use 'open' directly or 'combineReader'
  based on whether the parameter type is a list

trdthg commented on riscv-ctg/riscv_ctg/utils.py:
```diff
@@ -16,8 +16,14 @@
 yaml.allow_unicode = True
 
-def load_yaml(foo):
+def load_yaml(foo: str | list[str]):
```
I really want to write a type hint here to keep it clear, although the current code does not use type hints

trdthg replied:
finally deleted, because `|` is not supported in python 3.6

trdthg commented:
update

UmerShahidengr commented on riscv-ctg/riscv_ctg/utils.py:
```diff
@@ -60,9 +66,6 @@ def get_instr_list():
 
     return instr_lst
 
-def load_yamls(foo):
```
Thank you for removing this function

UmerShahidengr reviewed (approved):
LGTM
